app/views/samples_reports/show.haml

```diff
@@ -340,49 +347,25 @@
   async function printPDF() {
     var context = getUrlParameter('context');
     
-    var measured_signal_svg = encodeURIComponent(getSvg("barchart", "measured_signal"));
-    var specific_svg;
+    var params = {
+      id: #{@samples_report.id},
+      measured_signal_svg: encodeURIComponent(getSvg("barchart", "measured_signal"))
+    };
+
+    document.getElementById("print-pdf-measured-signal-svg").value = encodeURIComponent(getSvg("barchart", "measured_signal"));
+    
     if ("#{@purpose}" == "Challenge")
-      specific_svg = encodeURIComponent(getSvg("rocchart", "roc_curve"));
+      document.getElementById("print-pdf-specific-svg").value = encodeURIComponent(getSvg("rocchart", "roc_curve"));
     else
-      specific_svg = encodeURIComponent(getSvg("linechart", "limit_of_detection"));
+      document.getElementById("print-pdf-specific-svg").value = encodeURIComponent(getSvg("linechart", "roc_curve"));
```

Comment:
```suggestion
      document.getElementById("print-pdf-specific-svg").value = encodeURIComponent(getSvg("linechart", "limit_of_detection"));
```

Reply:
I'm also pretty sure you don't need to `encodeURIcomponent` the SVG. The browser will do that by itself (same on the backend: rails will decode automatically).